### PR TITLE
MNT Add pre-commit hooks

### PR DESCRIPTION
**Pull request by adrinjalali**
This PR adds pre-commit hooks to the repo.

These hooks are only enabled if the user installs them. By default cloning the repo will have them disabled.

Once enabled, they only run on changed files and not the whole repo which makes them quite fast.

cc @LysandreJik @osanseviero

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -198,8 +198,21 @@ Follow these steps to start contributing:
...
+   ```bash
+   pip install pre-commit
+   pre-commit install
+   ```
+
+   You only need to do the above once in your repository's environment. If for
+   any reason you would like to disable pre-commit hooks on a commit, you can
+   pass `-n` to your `git commit` command to temporarily disable pre-commit
+   hooks.
+
```
Comment (LysandreJik) [inline CONTRIBUTING.md:213]:
```suggestion
   You only need to do the above once in your repository's environment. If for
   any reason you would like to disable pre-commit hooks on a commit, you can
   pass `-n` to your `git commit` command to temporarily disable pre-commit
   hooks.
   
   To permanently disable hooks, you can run the following command:
   ```bash
   pre-commit uninstall
   ```
```

**Review by LysandreJik — APPROVED** [review]
LGTM, just added a minor suggestion

**Review by osanseviero — APPROVED** [review]
Nice, thanks!

**Comment by adrinjalali** [comment]
Merging with two approvals.

**Pull request merged**